online-shop/src/app/components/products-list/products-list.component.ts

```diff
@@ -0,0 +1,37 @@
+import {Component, OnInit} from '@angular/core';
+import {ProductModelDisplay} from "../../types/product-display.model";
+import {ProductService} from '../../services/product.service';
+import {OrderModel} from "../../types/order.model";
+import {UserService} from "../../services/user.service";
+
+@Component({
+  selector: 'app-products-list',
+  templateUrl: './products-list.component.html',
+  styleUrls: ['./products-list.component.scss']
+})
+export class ProductsListComponent implements OnInit {
+  products: ProductModelDisplay[] | undefined;
+  selectedProductID: string | undefined;
+  order: OrderModel = this.productService.getOrder();
+  hasAuthorisationOfAdmin: boolean = this.userService.hasRoleType("admin");
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`